File: sip-client/src/main/kotlin/com/basicfu/sip/client/common/FeignConfiguration.java
package com.basicfu.sip.client.common;

import com.basicfu.sip.client.util.RequestUtil;
import feign.RequestInterceptor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * @author basicfu
 * @date 2018/7/18
 */
@Configuration
public class FeignConfiguration {
    @Bean
    public RequestInterceptor requestInterceptor() {
        return requestTemplate -> {
            String header = RequestUtil.INSTANCE.getHeader(Constant.AUTHORIZATION);
            requestTemplate.header(Constant.AUTHORIZATION, header);
        };
    }
}

File: sip-client/src/main/kotlin/com/basicfu/sip/client/common/FeignConfiguration.kt
package com.basicfu.sip.client.common

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

/**
 * @author basicfu
 * @date 2018/7/18
 */
@Component
class FeignConfiguration {
    @Value("\${sip.app:}")
    private val app: String? = null
    @Value("\${sip.secret:}")
    private val secret: String? = null

}
